src/dbnode/storage/series/types.go

```diff
@@ -335,16 +335,28 @@ func (s Stats) IncCreatedEncoders() {
 type WriteType int
 
 const (
-	// UndefinedWriteType is an undefined write type.
-	UndefinedWriteType WriteType = iota
-
 	// WarmWrite represents warm writes (within the buffer past/future window).
-	WarmWrite
+	WarmWrite WriteType = iota
 
 	// ColdWrite represents cold writes (outside the buffer past/future window).
 	ColdWrite
 )
 
+// BootstrapWriteType is the write type assigned for bootstraps.
```

Comment:
Its not really arbitrary is it? I.E if you put data for a block that has already been flushed into the buffer and mark it as "warm" unless the regular flush cycle knows to account for that (which I don't think it currently does) something will get messed up I think (I think it won't get flushed because the regular flush cycle won't try to flush it)

Reply:
Hmm you're right, looks like we'd need more logic to keep track and include those blocks as part of the flush loop. Making it ColdWrite will avoid this issue since we'd be looking through the  entire range of possible blocks.

Reply:
True, while the time range the flush manager looks at is the entire retention period, we'd need additional logic to mark the flush state accordingly to signify that the block requires a flush. Setting this to ColdWrite may be simpler in this case, since all ColdWrites get persisted every compaction cycle.